Second tank

tymochko opened the pull request:
Please review our current progress for this week: 
- after player invite creates new game in DB
- redirects each of players (invitor and invited) to game page with data from DB
- instantiate Game instance for each player and draw tanks in each player's window

tymochko committed:
LVNODQIR-92 refactor client invites into dashboard controller

MykhailoDzedzei committed:
Added translation into Ukrainian language

MykhailoDzedzei committed:
LVNODQIR-103 translation of profile page

MykhailoDzedzei committed:
ESLint with new rules

MykhailoDzedzei committed:
fix about translate

tymochko committed:
LVNODQIR-92 implement service into game invites

tymochko committed:
Merge branch 'second_tank' of https://github.com/tymochko/pocket-tanks into second_tank

tymochko committed:
LVNODQIR-92 refactor service into object

OlgaLyubin committed:
LVNODQIR-101 Created route for scores page

tymochko committed:
LVNODQIR-92 implement game creation with embed users ids

tymochko committed:
LVNODQIR-92 working on simultanious redirect to game

- changes to eslint rules

kachanovskyi committed:
LVNODQIR-99: implemented tank class

kachanovskyi committed:
merged with second_tank

tymochko committed:
LVNODQIR-92 implement receive game data on init stage

kachanovskyi committed:
LVNODQIR-99: fixed tank shot

kachanovskyi committed:
LVNODQIR-99: merged with second_tank

kachanovskyi committed:
added player model

kachanovskyi committed:
Split home and navigation modules into files

tymochko committed:
LVNODQIR-92 fix originalPoints to receive filled in

MykhailoDzedzei committed:
LVNODQIR-106 created instantiate classes

MykhailoDzedzei committed:
resolved conflict inside gameCtrl

kachanovskyi committed:
fixed bullet hittin ground

MykhailoDzedzei committed:
LVNODQIR-106 added socket for transfer and update data into DB

kachanovskyi committed:
working on playersId

kachanovskyi committed:
Merge branch 'second_tank' into tank_movement

kachanovskyi committed:
merged with tank_movement

kachanovskyi committed:
Merge branch 'second_tank' into tank_movement

MykhailoDzedzei committed:
LVNODQIR-103 dashboard page translated

kachanovskyi committed:
LVNODQIR-99: working on second tank. Implemented different players IDs

tymochko committed:
Extend functionality for ES2015 in NodeJS

tymochko committed:
Merge branch 'second_tank' into service_gameData

tymochko committed:
refactoring socket on shot

tymochko committed:
refactoring socket on shot

MykhailoDzedzei committed:
translation fixes on Navigation

MykhailoDzedzei committed:
merge with second tank

maksym08 committed:
refactoring socket on shot

kachanovskyi committed:
LVNODQIR-99: two tanks on gamefield, working on split tank moves

maksym08 committed:
`Merge branch 'second_tank' into chat2

maksym08 committed:
Merge branch 'second_tank' into chat2

MykhailoDzedzei committed:
LVNODQIR-103 finished translation of the game pages

MykhailoDzedzei committed:
pull second tank

MykhailoDzedzei committed:
LVNODQIR-103 merge with translate_page

MykhailoDzedzei committed:
translation key loading on starting page

kachanovskyi committed:
LVNODQIR-99: tanks move in both windows

kachanovskyi committed:
merged with second_tank

maksym08 committed:
temp changes

maksym08 committed:
refactoring socket

Commit:
modal window for confirm

tymochko committed:
LVNODQIR-107 add button "surrender" and styles

Conflicts:
	src/client/models/shotTrajectory.js

kachanovskyi committed:
LVNODQIR-99: working on two tanks movement on both screens

kachanovskyi committed:
Merge branch 'second_tank' into tank_movement

tymochko committed:
LVNODQIR-107 refactor sockets

tymochko committed:
Merge branch 'second_tank' into service_gameData

Conflicts:
	src/client/models/externalFunctions.js

MykhailoDzedzei committed:
bullet intersect second tank

MykhailoDzedzei committed:
LVNODQIR-108 refactoring code and hide bullet after intersection

tymochko committed:
LVNODQIR-107 implement update in DB

tymochko committed:
Merge branch 'second_tank' into service_gameData

tymochko committed:
LVNODQIR-107 fix DB update

tymochko committed:
LVNODQIR-107 implement "SURRENDER" button

- redirects to Dashboard Page to both players
- fix Invitation modal to display user

kachanovskyi committed:
LVNODQIR-99: correct tank movement on both screens (needs refactor)

kachanovskyi committed:
LVNODQIR-99: merged with second tank

kachanovskyi committed:
LVNODQIR-99: fully merged with second_tank

tymochko committed:
LVNODQIR-109 implement player turn (move, shot)

- move chat button call to navPanel
- removed rule from eslint

kachanovskyi committed:
LVNODQIR-99: weapon moves on both screens respectively to the tanks

tymochko committed:
Merge branch 'second_tank' into service_gameData

Conflicts:
	src/client/models/externalFunctions.js
	src/client/models/tankShot.js

MykhailoDzedzei committed:
working on resume game

maksym08 committed:
merge second_tank

maksym08 committed:
merge second_tank

maksym08 committed:
fixed tank shot

MykhailoDzedzei committed:
LVNODQIR-112 resumme game after disconnect

maksym08 committed:
fixed chat

maksym08 committed:
fixed socket.js

kachanovskyi committed:
NavigationConstructorService modified

kachanovskyi committed:
Merge branch 'second_tank' into tank_movement

MykhailoDzedzei committed:
LVNODQIR-113 created socket for update data and LVNODQIR-112 resume game works

MykhailoDzedzei committed:
merge with second_tank

tymochko committed:
LVNODQIR-109 implement player highlight in turn

tymochko committed:
Merge branch 'second_tank' into service_gameData

Conflicts:
	src/client/models/tankShot.js
	src/server/socket/gameSocket.js

MykhailoDzedzei committed:
Fixed intersection bullet with tanks

MykhailoDzedzei committed:
merge with second tank

tymochko committed:
LVNODQIR-109 implement turn for weapon

tymochko committed:
LVNODQIR-109 implement change turn after battlefield damage

MykhailoDzedzei committed:
merge with second tank

maksym08 committed:
fixed ingame navpanel

maksym08 committed:
merge second_tank

maksym08 committed:
merge second_tank

MykhailoDzedzei committed:
fix turn

MykhailoDzedzei committed:
merge with second tank

olesiabrkt commented on src/client/languages/languages.js:
```diff
@@ -0,0 +1,101 @@
+    ErrorUserName2: " Ім'я користувача повинно складатися принаймні з 5 символів",
+    ErrorUserName3: 'Username cannot be longer than 15 characters',
+    ErrorUserAge1: 'Вам повинно бути не менше 18 років',
+    ErrorUserAge2: 'Ви не можете бути старше 100 років ',
```
=(((

olesiabrkt commented on src/client/models/externalFunctions.js:
```diff
@@ -35,3 +33,43 @@ export function fillBackground(ctx, pattern) {
+        thisWindowPlayerId = gameInst.player2.id;
+    }
+
+    if (thisWindowPlayerTurn === true) {
```
`if (thisWindowPlayerTurn) {` - the same

olesiabrkt commented on src/client/models/externalFunctions.js:
```diff
@@ -35,3 +33,43 @@ export function fillBackground(ctx, pattern) {
+    callback(tank2, tankImage, weaponImage);
+};
+
+export const allowTurn = (gameInst) => {
```
the name of the function says like it should return boolean for me...

olesiabrkt commented on src/client/models/externalFunctions.js:
```diff
@@ -35,3 +33,43 @@ export function fillBackground(ctx, pattern) {
+};
+
+export const changeTurn = (playerTurn) => {
+    if (playerTurn) {
```
haha, really? what for? just use `!playerTurn `

olesiabrkt commented on src/client/models/externalVariables.js:
```diff
@@ -1,8 +1,8 @@
+export const TANK_WIDTH = 70;
+export const TANK_HEIGHT = 30;
+export const WEAPON_WIDTH = 35;
+export const WEAPON_HEIGHT = 20;
```
nice

olesiabrkt commented on src/client/models/gameModel.js:
```diff
@@ -0,0 +1,20 @@
+
+
+
+// gameObj;
```
should be removed

olesiabrkt commented on src/client/models/navPanel.js:
```diff
@@ -1,58 +1,93 @@
-        power++;
-        getId('power').innerHTML = power;
         power = parseInt(getId('power').innerHTML);
+        power += 5;
```
what 5 means?

olesiabrkt commented on src/client/models/navPanel.js:
```diff
@@ -1,58 +1,93 @@
+            gameInst.player2.life = 0;
+        }
+
+        socket.emit('end-game', gameInst);
```
I hope server emits end game for the 2nd user, right?

olesiabrkt commented on src/client/models/shotTrajectory.js:
```diff
@@ -1,61 +1,78 @@
     lastTime,
-    dt2 = 0,
+    deltaT = 0,
+    deltaT_tank1 = 0,
```
do not use underscore with camelcase please

olesiabrkt commented on src/client/models/shotTrajectory.js:
```diff
@@ -1,61 +1,78 @@
-const shotStart = () => {
-
+const shotStart = (check_deltaT = true) => {
+	if(!check_deltaT) deltaT = deltaT_tank1;
```
forgot brackets

olesiabrkt commented on src/client/models/shotTrajectory.js:
```diff
@@ -64,39 +81,93 @@ const drawBullet = () => {
     //check angle for accuracy of point
-    bull.rotate(-bullet.imgInf.currAngle);
+    bull.rotate(-bullet.imgInf.currAngle * 180 / Math.PI);
+    bull.strokeColor = '#ff0000';
```
why have we to highlight it with red?

olesiabrkt commented on src/client/models/shotTrajectory.js:
```diff
@@ -64,39 +81,93 @@ const drawBullet = () => {
+
+    var tankSize = new paper.Size(80,30);
+    var user1 = new paper.Path.Rectangle(new paper.Point(player1.data.tankX - tankSize.width / 2, player1.data.tankY - tankSize.height / 2), tankSize);
+    user1.strokeColor = '#00ff00';
```
what these colors for?

olesiabrkt commented on src/client/models/shotTrajectory.js:
```diff
@@ -64,39 +81,93 @@ const drawBullet = () => {
     // check if intersect the original points
+    var intersectPlayer1 = bull.getIntersections(user2);
+    if(intersectPlayer1.length > 0) {
+        console.log('boom2');
```
ha! :)) remove it

olesiabrkt commented on src/client/models/shotTrajectory.js:
```diff
@@ -107,40 +178,35 @@ const generateExplosion = (dt) => {
+        socket.emit('update-data', gameData);
 
-        ground.setGround(calculatedGroundPoints);
+        // ground.setGround(calculatedGroundPoints);
```
just run lint :) I will not be pointing every single item like this.

mnkami commented on src/client/models/tankShot.js:
```diff
@@ -1,120 +1,267 @@
-        let now = new Date().getTime();
+        const now = new Date().getTime();
+
+        if ((now - lastTimeTankMoved > 800) && (allowTurn(gameInst) === localStorage.getItem('playerId'))) {
```
move 800 to some const

mnkami commented on src/client/models/tankShot.js:
```diff
@@ -1,120 +1,267 @@
+                    drawTanks(drawTank, tank1, tank2, tankImage, weaponImage);
+                    break;
 
+                default:
```
you could remove default as you are not performing anything

kachanovskyi replied:
sorry, ESLint wanted to have default case, that's why I wrote it.

mnkami commented on src/client/models/tankShot.js:
```diff
@@ -1,120 +1,267 @@
-    navPanel(angle, tankX, tankY, weaponAngle);
+    socket.on('outputPosTank', (data) => {
+        tankMove(data.direction, data.tankMoves, data.tank1, data.tank2, tankImage, weaponImage, socket);
+        console.log(allowTurn(gameInst), 'allowTurn');
```
still in progress?

tymochko replied:
Yes, sorry. We decided today is our final day for implementation functionality.

mnkami commented on src/client/modules/dashboard/index.js:
```diff
@@ -1,28 +1,77 @@
+    });
+
+    socket.on('invite-rejected', function(data) {
+        alert('Your invitation was rejected.');
```
should be a modal already

mnkami commented on src/client/modules/dashboard/requestToPlay.html:
```diff
@@ -0,0 +1,9 @@
+<div class="modal-header">
+</div>
+<div class="modal-body">
+    <h3 class="modal-title">Wanna play with  {{data.senderUsername}} ?</h3>
```
@MykhailoDzedzei translate

mnkami commented on src/client/modules/game/gameService.js:
```diff
@@ -0,0 +1,27 @@
+            const deferred = $q.defer();
+
+            socket.on('get-game-data', (gameData) => {
+                console.log('2');
```
remove

mnkami commented on src/client/modules/profile/ManageProfileController.js:
```diff
@@ -26,32 +26,52 @@ export class ManageProfileController {
-                closeHtml: '<button>&times;</button>'
-            });
+        function savingMsg() {
+            $translate('savingMsg')
```
use Promise.all

mnkami commented on src/client/modules/profile/ManageProfileController.js:
```diff
@@ -26,32 +26,52 @@ export class ManageProfileController {
-                closeButton: true,
-                closeHtml: '<button>&times;</button>'
+        function avatarMsg() {
+            $translate('avatarMsg')
```
Promise.all

mnkami commented on src/server/api/users/usersController.js:
```diff
@@ -280,7 +286,11 @@ const getUserImage = function (req, res) {
+        }
         if (userImage.uploadedImg) {
+
             userDir = __dirname + '/../../static/usersInfo/' + userId + '/' + userImage.image;
```
@MykhailoDzedzei you always forget about these lines!!! :)

mnkami commented on src/server/socket/chat.js:
```diff
@@ -0,0 +1,37 @@
+let mongoose = require('mongoose');
```
fix the code style

tymochko replied:
sorry, what do you mean?

mnkami commented on src/client/models/externalFunctions.js:
```diff
@@ -35,3 +33,43 @@ export function fillBackground(ctx, pattern) {
+    return null;
+};
+
+export const changeTurn = (playerTurn) => {
```
just `!playerTurn`

tymochko replied:
thanks. That's simple.

shmoksi committed:
fix reload page SignUp

shmoksi committed:
refactor code

shmoksi committed:
fix session after register

shmoksi committed:
fix redirect after logout

shmoksi committed:
fix style: remove white line before footer & keep footer to bottom

MykhailoDzedzei committed:
fixed socket for update data in DB and painting map synchronously on both screens

shmoksi committed:
add style for lifebar

shmoksi committed:
Merge branch 'second_tank' of https://github.com/tymochko/pocket-tanks into lifebar

shmoksi committed:
add logic for lifebar

tymochko committed:
LVNODQIR-109 refactor changeTurn function

tymochko committed:
LVNODQIR-109 Merge || fix turn highlight

Conflicts:
	src/client/models/externalFunctions.js
	src/client/models/shotTrajectory.js
	src/client/models/tankShot.js

OlgaLyubin committed:
LVNODQIR-101 scores page

OlgaLyubin committed:
LVNODQIR-101 scores page

shmoksi committed:
fix style

shmoksi committed:
Merge branch 'lifebar' into second_tank

tymochko committed:
Refactor shotTrajectory

tymochko committed:
Merge branch 'second_tank' into service_gameData

kachanovskyi committed:
refactored tankShot.js

kachanovskyi committed:
merged with second_tank

maksym08 committed:
fixed chat

maksym08 committed:
merge second_tank

kachanovskyi committed:
fixed logination (removed login and password checks)

shmoksi committed:
remove magic number & change color life

shmoksi committed:
Merge branch 'lifebar' into second_tank

# Conflicts:
#	src/client/models/externalVariables.js
